### PR TITLE
Add IDF component manifest and workflow

### PR DESCRIPTION
**Pull request by tore-espressif**
Hello @igrr,
This PR adds a workflow for automatic upload to Espressif's component service on push to main branch.

Please let me know whether the secret (API token for component service) gets into upstream repository, so I could send it to you in case it doesn't.

Feel free to change the component version in `idf_component.yml`.

### CODE REVIEW

File: .github/workflows/upload_component.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    steps:
+      - uses: actions/checkout@master
+        with:
+          submodules: "recursive"
+
+      - name: Upload components to component service
+        uses: espressif/github-actions/upload_components@master
+        with:
+          name: "libnmea"
+          namespace: "espressif"
```
Comment (igrr) [inline .github/workflows/upload_component.yml]: I think I'd prefer to keep this component under my personal namespace, not `espressif`. I'll check with @kumekay if creating such namespace is possible already.

**Review by igrr — COMMENTED** [review]
Thanks for the PR @tore-espressif!

File: .github/workflows/upload_component.yml (hunk 1 above)
Comment (kumekay) [reply .github/workflows/upload_component.yml]: Yes, it's possible but manually at the moment. What namespace do you what for personal projects?

Comment (igrr) [reply .github/workflows/upload_component.yml]: `igrr`, if it's not taken :) Thanks in advance

Comment (kumekay) [reply .github/workflows/upload_component.yml]: I've created the namespace (and empty component in it)

Comment (tore-espressif) [reply .github/workflows/upload_component.yml]: Namespace updated to `igrr`. 
API token updated too, to reflect namespace change

**Comment by tuupola** [comment]
Is there any documentation about the Espressif component service? Quick search reveals only the cli and registry site itself:

https://pypi.org/project/idf-component-manager/
https://components.espressif.com/

**Comment by kumekay** [comment]
@tuupola Thank you for the question. Component Manager is not publicly released yet, however, the beta announcement and public docs are coming in the next several days. Stay tuned

**Comment by tore-espressif** [comment]
@igrr could you please re-run the github action? it should be all fixed now

**Comment by igrr** [comment]
It don't see a "re-run workflow" button anymore, maybe it is disabled some time after the workflow was first run.
It will probably run again next time I push some commit to the repo.

**Pull request merged**